Source/DafnyCore/Rewriters/ExpectContracts.cs

```diff
@@ -119,49 +117,28 @@ private MemberDecl GenerateFunctionWrapper(TopLevelDeclWithMembers parent, Membe
 
     newFunc.Body = callExpr;
 
-    var localName = origFunc.Result?.Name ?? "__result";
-    var localExpr = new IdentifierExpr(tok, localName) {
-      Type = newFunc.ResultType
-    };
+    var resultVar = origFunc.Result ?? new Formal(tok, "#result", newFunc.ResultType, false, false, null);
+    var localExpr = Expression.CreateIdentExpr(resultVar);
 
     var callRhs = new ExprRhs(callExpr);
 
     var lhss = new List<Expression> { localExpr };
     var rhss = new List<AssignmentRhs> { callRhs };
 
-    var assignStmt = new SingleAssignStmt(decl.RangeToken, localExpr, callRhs);
-    Statement callStmt;
-    if (origFunc.Result?.Name is null) {
-      var local = new LocalVariable(decl.RangeToken, localName, newFunc.ResultType, false);
-      local.type = newFunc.ResultType;
-      var locs = new List<LocalVariable> { local };
-      var varDeclStmt = new VarDeclStmt(decl.RangeToken, locs, new AssignStatement(decl.RangeToken, lhss, rhss) {
-        ResolvedStatements = new List<Statement>() { assignStmt }
-      });
-      localExpr.Var = local;
-      callStmt = varDeclStmt;
-    } else {
-      localExpr.Var = origFunc.Result;
-      callStmt = assignStmt;
-    }
```

Comment:
Not completely following why this logic became unnecessary?

Reply:
Despite writing the previous code, I don't understand well either, but this was conditional on `origFunc.Result == null` and above I do `var resultVar = origFunc.Result ?? new Formal(tok, "#result", newFunc.ResultType, false, false, null);` which handles the same thing.